flashinfer/jit/core.py

```diff
@@ -197,6 +197,10 @@ def gen_jit_spec(
         # non debug mode
         cuda_cflags += ["-DNDEBUG"]
 
+    # useful for ncu
+    if bool(os.environ.get("FLASHINFER_JIT_LINEINFO", "0")):
```

Comment:
`bool("0")` evaluates to `True`, so `bool(os.environ.get("FLASHINFER_JIT_LINEINFO", "0"))` will be True if this environment variable is not set.

This bug is fixed in #1872 , note that `-lineinfo` will greatly increase binary size, see my comment in https://github.com/flashinfer-ai/flashinfer/pull/1872#issuecomment-3374843208

cc @fzyzcjy @zhyncs  for viz

Reply:
Good point, I should have written `bool(int(...))`...